### PR TITLE
[WIP] update env var doc

### PR DESCRIPTION
**Pull request by rosieyohannan**
some general updates to env vars doc

- [x] ui screenshots for env vars and contexts (undecided if these are necessary)
- [x] order of precedence diagram
- [x] fixed order of precedence which was slightly wrong
- [x] updated examples to use 2.1 config
- [x] corrected main example to show all methods and that only project and context env vars are masked using secrets masking
- [x] added example of injecting variables with API v2 (pipeline parameters)

### CODE REVIEW

File: jekyll/_cci2/env-vars.md [hunk 1]
```diff
@@ -243,17 +287,66 @@ jobs:
...
+          command: |
+            echo $MY_ENV_VAR
+```
+
+Creating a context allows you to share environment variables across multiple projects, and control who has access. For more information, see the [Contexts documentation]({{ site.baseurl }}/2.0/contexts/).
 
 ## Setting an Environment Variable in a Project
 
-1. In the CircleCI application, go to your project's settings by clicking the gear icon next to your project.
+1. In the CircleCI application, go to your project's settings by clicking the gear icon on the Pipelines page, por the three dots on other pages in the application.
```
Comment (thomasr8) [inline jekyll/_cci2/env-vars.md]:
```suggestion
1. In the CircleCI application, go to your project's settings by clicking the gear icon on the Pipelines page, or the three dots on other pages in the application.
```

File: jekyll/_api/v1-reference.md [hunk 2]
```diff
@@ -209,7 +209,7 @@ You can retry a build with ssh by swapping "retry" with "ssh":
 
 <h2 id="new-build-branch">Trigger a new Build with a Branch</h2>
 
-<span class='label label-info'>Note:</span> For more about build parameters, read about [using 1.0 parameterized builds]( {{ site.baseurl }}/1.0/parameterized-builds/) and [optional 2.0 build parameters]({{ site.baseurl }}/2.0/env-vars/#injecting-environment-variables-with-the-api). The response for "failed" should be a boolean `true` or `null`.
+<span class='label label-info'>Note:</span> For more about build parameters, read about [using 1.0 parameterized builds]( {{ site.baseurl }}/1.0/parameterized-builds/) and [optional 2.0 build parameters]({{ site.baseurl }}/2.0/env-vars/#injecting-environment-variables-with-api-v1). The response for "failed" should be a boolean `true` or `null`.
```
Comment (thomasr8) [inline jekyll/_api/v1-reference.md]:
```suggestion
<span class='label label-info'>Note:</span> For more information about build parameters, refer to the [using 1.0 parameterized builds]( {{ site.baseurl }}/1.0/parameterized-builds/) and [optional 2.0 build parameters]({{ site.baseurl }}/2.0/env-vars/#injecting-environment-variables-with-api-v1) pages. The response for "failed" should be a boolean `true` or `null`.
```

File: jekyll/_cci2/env-vars.md [hunk 3]
```diff
@@ -15,92 +15,112 @@ This document describes using environment variables in CircleCI in the following
...
+There are several ways to use environment variables in CircleCI to provide variety in scope and authorization level. Environment variables are governed by an [order of precedence](#order-of-precedence), depending on how they are set, allowing control at each level in your configuration. 
 
-### Secrets Masking
-{:.no_toc}
+To add **private keys** or **secret environment variables** for use throughout your private project, use the [Environment Variables page under Project Settings](#setting-an-environment-variable-in-a-project) in the CircleCI application. The variable values are neither readable nor editable in the app after they are set. To change the value of an environment variable, delete the current variable and add it again with the new value.
+
+Private environment variables enable you to store secrets safely even when your project is public, see [Building Open Source Projects]({{ site.baseurl }}/2.0/oss/) for associated settings information.
```
Comment (thomasr8) [inline jekyll/_cci2/env-vars.md]:
```suggestion
Private environment variables enable you to store secrets safely even when your project is public. Refer to the [Building Open Source Projects]({{ site.baseurl }}/2.0/oss/) page for associated settings information. 
```

File: jekyll/_cci2/env-vars.md [hunk 4]
```diff
@@ -15,92 +15,112 @@ This document describes using environment variables in CircleCI in the following
...
 
-### Secrets Masking
-{:.no_toc}
+To add **private keys** or **secret environment variables** for use throughout your private project, use the [Environment Variables page under Project Settings](#setting-an-environment-variable-in-a-project) in the CircleCI application. The variable values are neither readable nor editable in the app after they are set. To change the value of an environment variable, delete the current variable and add it again with the new value.
+
+Private environment variables enable you to store secrets safely even when your project is public, see [Building Open Source Projects]({{ site.baseurl }}/2.0/oss/) for associated settings information. 
+
+Use Contexts to [further restrict access to environment variables](#setting-an-environment-variable-in-a-context). Contexts are set from the Organization Settings in the CircleCI application. For more information refer to the [Restricting a Context]({{ site.baseurl }}/2.0/contexts/#restricting-a-context) documentation.
```
Comment (thomasr8) [inline jekyll/_cci2/env-vars.md]:
```suggestion
Use Contexts to [further restrict access to environment variables](#setting-an-environment-variable-in-a-context). Contexts are set from the Organization Settings in the CircleCI application. For more information about setting setting contexts, refer to the [Restricting a Context]({{ site.baseurl }}/2.0/contexts/#restricting-a-context) documentation.
```

File: jekyll/_cci2/env-vars.md [hunk 5]
```diff
@@ -15,92 +15,112 @@ This document describes using environment variables in CircleCI in the following
...
 The above `config.yml` demonstrates the following: 
 
 - Setting custom environment variables
 - Reading a built-in environment variable that CircleCI provides (`CIRCLE_BRANCH`)
 - How variables are used (or interpolated) in your `config.yml`
-- Masking of printed environment variables (secrets masking)
+- Secrets masking, applied to environment variable set in the project or within a Context.
 
-When the above config runs, the output looks like this:
+When the above config runs, the output looks like this, notice the env var stored in the Project is masked, and displays as `****`:
```
Comment (thomasr8) [inline jekyll/_cci2/env-vars.md]:
```suggestion
When the above config runs, the output looks like this. Notice the env var stored in the Project is masked, and displays as `****`:
```

File: jekyll/_cci2/env-vars.md [hunk 6]
```diff
@@ -111,9 +131,26 @@ In the example below, `$ORGNAME` and `$REPONAME` will not be interpolated.
...
+    repo_name:
+      type: string
+      default: my_repo
+  my_job:
+    docker:
+      - circleci/node:12.15.0
+    working_directory: /go/src/github.com/<< parameters.org_name >>/<< parameters.repo_name >>
+```
+
+Consider reading the documentation on [using the parameters declaration]({{ site.baseurl }}/2.0/reusing-config/#using-the-parameters-declaration).
```
Comment (thomasr8) [inline jekyll/_cci2/env-vars.md]:
```suggestion
For more information, read the documentation on [using the parameters declaration]({{ site.baseurl }}/2.0/reusing-config/#using-the-parameters-declaration).
```

File: src-api/source/includes/_builds_and_jobs.md [hunk 7]
```diff
@@ -376,7 +376,7 @@ https://circleci.com/api/v1.1/project/:vcs-type/:username/:project/tree/:branch?
 **`POST` Request:** Triggers a new build and then returns a summary of the build.
 
 <aside class="notice">
-For more about build parameters, see 2.0 build parameters <a href="https://circleci.com/docs/2.0/env-vars/#injecting-environment-variables-with-the-api">for environment variables</a>. The response for “failed” should be a boolean true or null.
+For more about build parameters, see 2.0 build parameters <a href="https://circleci.com/docs/2.0/env-vars/#injecting-environment-variables-with-api-v1">for environment variables</a>. The response for “failed” should be a boolean true or null.
```
Comment (thomasr8) [inline src-api/source/includes/_builds_and_jobs.md]:
```suggestion
For more about build parameters, see 2.0 build parameters <a href="https://circleci.com/docs/2.0/env-vars/#injecting-environment-variables-with-api-v1">for environment variables</a>. The response for “failed” should be a boolean `true` or `null`.
```

File: jekyll/_cci2/env-vars.md (hunk 4 above)
Comment (rosieyohannan) [reply jekyll/_cci2/env-vars.md]: I think I'll change this to "for more information about controlling access to env vars using contexts. ..."

File: jekyll/_cci2/env-vars.md [hunk 8]
```diff
@@ -15,92 +15,112 @@ This document describes using environment variables in CircleCI in the following
...
 
-### Secrets Masking
-{:.no_toc}
+To add **private keys** or **secret environment variables** for use throughout your private project, use the [Environment Variables page under Project Settings](#setting-an-environment-variable-in-a-project) in the CircleCI application. The variable values are neither readable nor editable in the app after they are set. To change the value of an environment variable, delete the current variable and add it again with the new value.
+
+Private environment variables enable you to store secrets safely even when your project is public. Refer to the [Building Open Source Projects]({{ site.baseurl }}/2.0/oss/) page for associated settings information. 
+
+Use Contexts to [further restrict access to environment variables](#setting-an-environment-variable-in-a-context). Contexts are set from the Organization Settings in the CircleCI application. For more information refer to the [Restricting a Context]({{ site.baseurl }}/2.0/contexts/#restricting-a-context) documentation.
```
Comment (rosieyohannan) [inline jekyll/_cci2/env-vars.md]:
```suggestion
Use Contexts to [further restrict access to environment variables](#setting-an-environment-variable-in-a-context). Contexts are set from the Organization Settings in the CircleCI application. For more information about controlling access to env vars with Contexts, refer to the [Restricting a Context]({{ site.baseurl }}/2.0/contexts/#restricting-a-context) documentation.
```

File: jekyll/_cci2/env-vars.md [hunk 9]
```diff
@@ -133,26 +170,29 @@ allowing you to use interpolation and share environment variables across `run` s
...
-
     shell: /bin/sh -leo pipefail
     environment:
       - BASH_ENV: /etc/profile
 ```
 
 ## Setting an Environment Variable in a Shell Command
 
-While CircleCI does not support interpolation when setting environment variables, it is possible to set variables for the current shell by [using `BASH_ENV`](#using-bash_env-to-set-environment-variables). This is useful for both modifying your `PATH` and setting environment variables that reference other variables.
+While CircleCI does not support interpolation when setting environment variables, it is possible to set variables for the current shell by [using `BASH_ENV`](#using-parameters-and-bashenv). This is useful for both modifying your `PATH` and setting environment variables that reference other variables.
```
Comment (teesloane) [inline jekyll/_cci2/env-vars.md]:
```suggestion
While CircleCI does not support interpolation when setting environment variables, it is possible to set variables for the current shell by [using `BASH_ENV`](#using-parameters-and-bash_env). This is useful for both modifying your `PATH` and setting environment variables that reference other variables.
```

Comment (rosieyohannan) [reply jekyll/_cci2/env-vars.md]: ohhhh right! Thanks so much for this @teesloane it's been doing my head in. Will see if it works :-)

**Pull request merged**